### PR TITLE
axom: add spack test

### PR DESCRIPTION
**Pull request by balos1**
Adds standalone tests and build-time test.

### CODE REVIEW

**Comment by balos1** [comment]
@spackbot fix style

**Comment by balos1** [comment]
Eh, looks like CI is disabled for draft PRs. Ill mark it as ready now. @white238 Should we get this merged in the package.py in the axom repo before merging it here?

**Comment by balos1** [comment]
Im wondering if the CI runs the tests... looks like it does

File: var/spack/repos/builtin/packages/axom/package.py
Comment (scheibelp) [inline var/spack/repos/builtin/packages/axom/package.py:619]: (question) what is the purpose of this method?

Comment (balos1) [reply var/spack/repos/builtin/packages/axom/package.py:619]: To run the installation smoke tests with `spack test`.

**Review by scheibelp — APPROVED** [review]

**Comment by alecbcs** [comment]
@scheibelp is this good to merge? / Are we waiting on @white238?

**Comment by balos1** [comment]
@alecbcs @white238 should approve first

**Comment by scheibelp** [comment]
@alecbcs when I review a package PR where the package has a maintainer, I usually wait 5ish days for a review from them; if there is no response, I merge it myself. For this PR I set up a calendar reminder so as not to lose track of it. 

Since @balos1 explicitly mentioned we should wait, I will wait indefinitely though.

(also FWIW you and I were talking about merge policies in slack and I should say that this is my own personal algorithm, if you want to merge PRs sooner, then feel free)

**Review by white238 — APPROVED** [review]
Awesome! Thanks for taking care of this @balos1

**Pull request merged**
